benchmark/gen_data.py

```diff
@@ -456,15 +476,23 @@ def make_classification_udf(iter: Iterator[pd.Series]) -> pd.DataFrame:
                 .drop(*feature_cols)
             )
 
+        train_df, eval_df = df.randomSplit(
+            [args.train_size, 1 - args.train_size], seed=1
+        )
+
         if args.output_num_files is not None:
-            df = df.repartition(args.output_num_files)
+            train_df = train_df.repartition(args.output_num_files)
+            eval_df = eval_df.repartition(args.output_num_files)
 
-        df.printSchema()
+        train_df.printSchema()
 
-        writer = df.write
+        train_writer = train_df.write
+        eval_writer = eval_df.write
         if args.overwrite:
-            writer = df.write.mode("overwrite")
+            train_writer = train_df.write.mode("overwrite")
+            eval_writer = eval_df.write.mode("overwrite")
 
-        writer.parquet(args.output_dir)
+        train_writer.parquet(f"{args.output_dir}/train")
+        eval_writer.parquet(f"{args.output_dir}/eval")
```

Comment:
Can you also update `run_benchmark.sh` to use this new datagen (and update the various `--transform_path` args accordingly)?  (Could be in a followup PR)